### PR TITLE
Testsuite improvements

### PR DESCRIPTION
**Pull request by daneryl**
fixes: #3561

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

File: app/api/files/specs/jsRoutes.spec.js [hunk 1]
```diff
@@ -26,8 +27,9 @@ describe('upload routes', () => {
   let req;
   let file;
 
-  const deleteAllFiles = cb => {
-    const directory = `${__dirname}/uploads/`;
+  const deleteAllFiles = async cb => {
```
Comment (RafaPolit) [inline app/api/files/specs/jsRoutes.spec.js:30]: There are more calls to `deleteAllFiles` in this spec that are not being awaited.  They have a `done` inside that could cause the same effect, but maybe for consistency add the await on those calls? (currently lines 88 and 186)

File: app/api/files/filesystem.ts [hunk 2]
```diff
@@ -47,12 +47,34 @@ async function deleteFiles(files: FilePath[]) {
...
+
+const generateUploadsPath = async (subPath: string) => {
+  if (subPath) {
+    await createDirIfNotExists(`${__dirname}/specs/uploads/${subPath}`);
+  }
+  return {
+    uploadedDocuments: `${__dirname}/specs/uploads/${subPath}`,
+    attachments: `${__dirname}/specs/uploads/${subPath}`,
+    customUploads: `${__dirname}/specs/uploads/${subPath}`,
+    temporalFiles: `${__dirname}/specs/uploads/${subPath}`,
```
Comment (RafaPolit) [inline app/api/files/filesystem.ts:72]: Same concern as above,! Maybe this solves it if we are using only one at a time in each test?

**Review by RafaPolit — COMMENTED** [review]
I think these are great improvements.  Just a couple of comments / questions more than changes request.   I ran the suite several times, I had the `exportRoutes` timeout in one of the runs, but could not replicate it again, so it may have been an intermittent test or something just hanging up on my end.

File: app/api/files/filesystem.ts [hunk 3]
```diff
@@ -47,12 +47,34 @@ async function deleteFiles(files: FilePath[]) {
 const testingUploadPaths = {
   uploadedDocuments: `${__dirname}/specs/uploads/`,
   attachments: `${__dirname}/specs/uploads/`,
-  customUploads: `${__dirname}/specs/customUploads/`,
+  customUploads: `${__dirname}/specs/uploads/`,
   temporalFiles: `${__dirname}/specs/uploads/`,
```
[the remark this answers is not in the document]
Comment (daneryl) [reply app/api/files/filesystem.ts]: @RafaPolit i have reverted the customUploads path, its now /specs/customUploads again, for the others i think the issue described still remains, we should create a tech debt.

File: app/api/files/specs/jsRoutes.spec.js (hunk 1 above)
Comment (daneryl) [reply app/api/files/specs/jsRoutes.spec.js:31]: Good catch, the 2 calls are now properly awaited

File: app/api/files/filesystem.ts (hunk 2 above)
Comment (daneryl) [reply app/api/files/filesystem.ts]: this was not being used anymore actually, it has been removed now.

**Review by RafaPolit — APPROVED** [review]

**Comment by RafaPolit** [comment]
LGTM

**Pull request merged**